src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -23,17 +24,21 @@ internal sealed partial class AuthHandshakeMessageHandler : DelegatingHandler
 {
     private const int MaxRequestRetries = 5; // Arbitrary but seems to work ok for chunked uploads to ghcr.io
 
-    private sealed record AuthInfo(Uri Realm, string Service, string? Scope);
+    private sealed record AuthInfo(string Realm, string? Service, string? Scope);
+
+    private readonly string _registryName;
+
+    private AuthenticationHeaderValue? _authenticationHeader = null;
```

Comment:
I took the liberty of removing the `AuthHeaderCache` in favor of caching a single auth header in the handler.
Let me know if this is not ok, and I'll revert the change.

Reply:
I think we need to be able to handle multiple cached authentications here - both the source registry for the base image and the destination registry for the generated image could require auth.

Reply:
I've added back the shared cache. It now uses `registryName` as the key.